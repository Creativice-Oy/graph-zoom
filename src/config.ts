import {
  IntegrationExecutionContext,
  IntegrationValidationError,
  IntegrationInstanceConfigFieldMap,
  IntegrationInstanceConfig,
} from '@jupiterone/integration-sdk-core';
import { createAPIClient } from './client';

export const instanceConfigFields: IntegrationInstanceConfigFieldMap = {
  zoomAccessToken: {
    type: 'string',
    mask: true,
  },
  scopes: {
    type: 'string',
  },
};

export interface IntegrationConfig extends IntegrationInstanceConfig {
  /**
   * The OAuth token used to authenticate requests.
   */
  zoomAccessToken: string;
  scopes: string;
}

export async function validateInvocation(
  context: IntegrationExecutionContext<IntegrationConfig>,
) {
  const { config } = context.instance;

  if (!config.zoomAccessToken || !config.scopes) {
    throw new IntegrationValidationError(
      'Config requires zoom access token and scopes field. Please generate from OAuth server.',
    );
  }

  const apiClient = createAPIClient(config);
  await apiClient.verifyAuthentication();
}
