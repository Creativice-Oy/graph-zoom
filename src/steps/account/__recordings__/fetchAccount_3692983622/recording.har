{
  "log": {
    "_recordingName": "fetchAccount",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "4.3.0"
    },
    "entries": [
      {
        "_id": "32b6c540dbe7339d81f7948939897f76",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "*/*"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "api.zoom.us"
            }
          ],
          "headersSize": 862,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://api.zoom.us/v2/users/me"
        },
        "response": {
          "bodySize": 739,
          "content": {
            "mimeType": "application/json;charset=UTF-8",
            "size": 739,
            "text": "{\"id\":\"H9Yx0-O8QUCXVIseYAsjaA\",\"first_name\":\"JupiterOne\",\"last_name\":\"Development\",\"email\":\"development@jupiterone.dev\",\"type\":2,\"role_name\":\"Owner\",\"pmi\":4440405982,\"use_pmi\":false,\"personal_meeting_url\":\"https://zoom.us/j/4440405982?pwd=c0Jidml0ZS9ZRzA3dEZWSTNHNHlPZz09\",\"timezone\":\"America/New_York\",\"verified\":1,\"dept\":\"\",\"created_at\":\"2021-10-27T00:00:49Z\",\"last_login_time\":\"2021-11-02T09:51:57Z\",\"host_key\":\"815201\",\"cms_user_id\":\"\",\"jid\":\"h9yx0-o8qucxviseyasjaa@xmpp.zoom.us\",\"group_ids\":[\"ppUAdwEBRFCdRE0mfLttog\"],\"im_group_ids\":[],\"account_id\":\"N0oChUdLTrCZLfk9JXp3Hw\",\"language\":\"\",\"phone_country\":\"\",\"phone_number\":\"\",\"status\":\"active\",\"job_title\":\"\",\"location\":\"\",\"login_types\":[100],\"role_id\":\"0\",\"account_number\":7011249145}"
          },
          "cookies": [
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "zm_aid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "zm_haid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "zm_tmaid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "zm_htmaid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "httpOnly": true,
              "name": "cred",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "_zm_ctaid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "_zm_chtaid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            }
          ],
          "headers": [
            {
              "name": "date",
              "value": "Wed, 03 Nov 2021 04:43:44 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json;charset=UTF-8"
            },
            {
              "name": "content-length",
              "value": "739"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "x-zm-trackingid",
              "value": "v=2.0;clid=aw1;rid=WEB_6b57aea1eaef35fe8e753d8acb99267a"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "cache-control",
              "value": "no-cache, no-store, must-revalidate, no-transform"
            },
            {
              "name": "pragma",
              "value": "no-cache"
            },
            {
              "name": "expires",
              "value": "Thu, 01 Jan 1970 00:00:00 GMT"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "name": "vary",
              "value": "Origin, Access-Control-Request-Method, Access-Control-Request-Headers"
            },
            {
              "name": "x-ratelimit-category",
              "value": "Light"
            },
            {
              "name": "x-zm-zoneid",
              "value": "VA2"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubDomains"
            },
            {
              "name": "x-xss-protection",
              "value": "1; mode=block"
            },
            {
              "name": "referrer-policy",
              "value": "strict-origin-when-cross-origin"
            }
          ],
          "headersSize": 1347,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-11-03T04:43:43.081Z",
        "time": 1176,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 1176
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
