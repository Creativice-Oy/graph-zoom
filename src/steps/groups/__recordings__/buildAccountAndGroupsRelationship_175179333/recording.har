{
  "log": {
    "_recordingName": "buildAccountAndGroupsRelationship",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "4.3.0"
    },
    "entries": [
      {
        "_id": "32b6c540dbe7339d81f7948939897f76",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "*/*"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "api.zoom.us"
            }
          ],
          "headersSize": 862,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://api.zoom.us/v2/users/me"
        },
        "response": {
          "bodySize": 739,
          "content": {
            "mimeType": "application/json;charset=UTF-8",
            "size": 739,
            "text": "{\"id\":\"H9Yx0-O8QUCXVIseYAsjaA\",\"first_name\":\"JupiterOne\",\"last_name\":\"Development\",\"email\":\"development@jupiterone.dev\",\"type\":2,\"role_name\":\"Owner\",\"pmi\":4440405982,\"use_pmi\":false,\"personal_meeting_url\":\"https://zoom.us/j/4440405982?pwd=c0Jidml0ZS9ZRzA3dEZWSTNHNHlPZz09\",\"timezone\":\"America/New_York\",\"verified\":1,\"dept\":\"\",\"created_at\":\"2021-10-27T00:00:49Z\",\"last_login_time\":\"2021-11-02T09:51:57Z\",\"host_key\":\"815201\",\"cms_user_id\":\"\",\"jid\":\"h9yx0-o8qucxviseyasjaa@xmpp.zoom.us\",\"group_ids\":[\"ppUAdwEBRFCdRE0mfLttog\"],\"im_group_ids\":[],\"account_id\":\"N0oChUdLTrCZLfk9JXp3Hw\",\"language\":\"\",\"phone_country\":\"\",\"phone_number\":\"\",\"status\":\"active\",\"job_title\":\"\",\"location\":\"\",\"login_types\":[100],\"role_id\":\"0\",\"account_number\":7011249145}"
          },
          "cookies": [
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "zm_aid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "zm_haid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "zm_tmaid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "zm_htmaid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "httpOnly": true,
              "name": "cred",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "_zm_ctaid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "_zm_chtaid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            }
          ],
          "headers": [
            {
              "name": "date",
              "value": "Wed, 03 Nov 2021 05:00:24 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json;charset=UTF-8"
            },
            {
              "name": "content-length",
              "value": "739"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "x-zm-trackingid",
              "value": "v=2.0;clid=aw1;rid=WEB_ff81e1283a2aa139c3bb109408d22338"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "cache-control",
              "value": "no-cache, no-store, must-revalidate, no-transform"
            },
            {
              "name": "pragma",
              "value": "no-cache"
            },
            {
              "name": "expires",
              "value": "Thu, 01 Jan 1970 00:00:00 GMT"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "name": "vary",
              "value": "Origin, Access-Control-Request-Method, Access-Control-Request-Headers"
            },
            {
              "name": "x-ratelimit-category",
              "value": "Light"
            },
            {
              "name": "x-zm-zoneid",
              "value": "VA2"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubDomains"
            },
            {
              "name": "x-xss-protection",
              "value": "1; mode=block"
            },
            {
              "name": "referrer-policy",
              "value": "strict-origin-when-cross-origin"
            }
          ],
          "headersSize": 1347,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-11-03T05:00:23.589Z",
        "time": 1730,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 1730
        }
      },
      {
        "_id": "7892a632e9fd0e489b37b35fcbd815a2",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "*/*"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "api.zoom.us"
            }
          ],
          "headersSize": 860,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://api.zoom.us/v2/groups"
        },
        "response": {
          "bodySize": 258,
          "content": {
            "mimeType": "application/json;charset=UTF-8",
            "size": 258,
            "text": "{\"total_records\":2,\"groups\":[{\"id\":\"ppUAdwEBRFCdRE0mfLttog\",\"name\":\"Test Group\",\"total_members\":2,\"in_directory\":false,\"directory_privacy\":2},{\"id\":\"dSHPJbpfQo6uS1a0_iznyg\",\"name\":\"Test Group 2\",\"total_members\":0,\"in_directory\":false,\"directory_privacy\":2}]}"
          },
          "cookies": [
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "zm_aid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "zm_haid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "zm_tmaid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "zm_htmaid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "httpOnly": true,
              "name": "cred",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "_zm_ctaid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "_zm_chtaid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            }
          ],
          "headers": [
            {
              "name": "date",
              "value": "Wed, 03 Nov 2021 05:00:26 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json;charset=UTF-8"
            },
            {
              "name": "content-length",
              "value": "258"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "x-zm-trackingid",
              "value": "v=2.0;clid=aw1;rid=WEB_f0a5fb42877dd0301eede101d30665f1"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "cache-control",
              "value": "no-cache, no-store, must-revalidate, no-transform"
            },
            {
              "name": "pragma",
              "value": "no-cache"
            },
            {
              "name": "expires",
              "value": "Thu, 01 Jan 1970 00:00:00 GMT"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "name": "vary",
              "value": "Origin, Access-Control-Request-Method, Access-Control-Request-Headers"
            },
            {
              "name": "x-ratelimit-category",
              "value": "Medium"
            },
            {
              "name": "x-zm-zoneid",
              "value": "VA2"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubDomains"
            },
            {
              "name": "x-xss-protection",
              "value": "1; mode=block"
            },
            {
              "name": "referrer-policy",
              "value": "strict-origin-when-cross-origin"
            }
          ],
          "headersSize": 1348,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-11-03T05:00:25.328Z",
        "time": 905,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 905
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
