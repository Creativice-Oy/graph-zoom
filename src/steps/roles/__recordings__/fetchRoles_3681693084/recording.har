{
  "log": {
    "_recordingName": "fetchRoles",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "4.3.0"
    },
    "entries": [
      {
        "_id": "0b6d856aa89cbe61f82ab25f5324f4ef",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "*/*"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "api.zoom.us"
            }
          ],
          "headersSize": 859,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://api.zoom.us/v2/roles"
        },
        "response": {
          "bodySize": 448,
          "content": {
            "mimeType": "application/json;charset=UTF-8",
            "size": 448,
            "text": "{\"total_records\":3,\"roles\":[{\"id\":\"0\",\"name\":\"Owner\",\"description\":\"Account owner has full privileges to access and manage a Zoom account.\",\"total_members\":1},{\"id\":\"1\",\"name\":\"Admin\",\"description\":\"Admins have wide range privileges to access and manage a Zoom account.\",\"total_members\":0},{\"id\":\"2\",\"name\":\"Member\",\"description\":\"Members have access to basic Zoom video meeting functions but no account management privileges.\",\"total_members\":1}]}"
          },
          "cookies": [
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "zm_aid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "zm_haid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "zm_tmaid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "zm_htmaid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "httpOnly": true,
              "name": "cred",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "_zm_ctaid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "_zm_chtaid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            }
          ],
          "headers": [
            {
              "name": "date",
              "value": "Thu, 28 Oct 2021 10:18:08 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json;charset=UTF-8"
            },
            {
              "name": "content-length",
              "value": "448"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "x-zm-trackingid",
              "value": "v=2.0;clid=aw1;rid=WEB_fdba951d4fd8cf0f457e0f8f1d593ebd"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "cache-control",
              "value": "no-cache, no-store, must-revalidate, no-transform"
            },
            {
              "name": "pragma",
              "value": "no-cache"
            },
            {
              "name": "expires",
              "value": "Thu, 01 Jan 1970 00:00:00 GMT"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "name": "vary",
              "value": "Origin, Access-Control-Request-Method, Access-Control-Request-Headers"
            },
            {
              "name": "x-ratelimit-category",
              "value": "Medium"
            },
            {
              "name": "x-zm-zoneid",
              "value": "VA2"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubDomains"
            },
            {
              "name": "x-xss-protection",
              "value": "1; mode=block"
            },
            {
              "name": "referrer-policy",
              "value": "strict-origin-when-cross-origin"
            }
          ],
          "headersSize": 1348,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-10-28T10:18:07.607Z",
        "time": 1249,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 1249
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
