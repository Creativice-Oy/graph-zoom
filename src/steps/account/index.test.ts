import {
  createMockStepExecutionContext,
  Recording,
} from '@jupiterone/integration-sdk-testing';
import { IntegrationConfig } from '../../config';
import { fetchAccount } from '.';
import { integrationConfig } from '../../../test/config';
import { setupZoomRecording } from '../../../test/recording';

describe('#fetchAccount', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupZoomRecording({
      directory: __dirname,
      name: 'fetchAccount',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('should collect data', async () => {
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: integrationConfig,
    });

    await fetchAccount(context);

    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      collectedEntities: context.jobState.collectedEntities,
      encounteredTypes: context.jobState.encounteredTypes,
    }).toMatchSnapshot();

    const users = context.jobState.collectedEntities.filter((e) =>
      e._type.includes('zoom_account'),
    );
    expect(users.length).toBeGreaterThan(0);
    expect(users).toMatchGraphObjectSchema({
      _class: ['User'],
      schema: {
        additionalProperties: false,
        properties: {
          _rawData: {
            type: 'array',
            items: { type: 'object' },
          },
          _type: { const: 'zoom_account' },
          name: { type: 'string' },
          username: { type: 'string' },
          email: { type: 'string' },
        },
      },
    });
  });
});
