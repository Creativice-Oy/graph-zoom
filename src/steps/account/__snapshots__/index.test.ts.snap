// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#fetchAccount should collect data 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "User",
      ],
      "_key": "zoom_account:H9Yx0-O8QUCXVIseYAsjaA",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "account_id": "N0oChUdLTrCZLfk9JXp3Hw",
            "account_number": 7011249145,
            "cms_user_id": "",
            "created_at": "2021-10-27T00:00:49Z",
            "dept": "",
            "email": "development@jupiterone.dev",
            "first_name": "JupiterOne",
            "group_ids": Array [
              "ppUAdwEBRFCdRE0mfLttog",
            ],
            "host_key": "815201",
            "id": "H9Yx0-O8QUCXVIseYAsjaA",
            "im_group_ids": Array [],
            "jid": "h9yx0-o8qucxviseyasjaa@xmpp.zoom.us",
            "job_title": "",
            "language": "",
            "last_login_time": "2021-11-02T09:51:57Z",
            "last_name": "Development",
            "location": "",
            "login_types": Array [
              100,
            ],
            "personal_meeting_url": "https://zoom.us/j/4440405982?pwd=c0Jidml0ZS9ZRzA3dEZWSTNHNHlPZz09",
            "phone_country": "",
            "phone_number": "",
            "pmi": 4440405982,
            "role_id": "0",
            "role_name": "Owner",
            "status": "active",
            "timezone": "America/New_York",
            "type": 2,
            "use_pmi": false,
            "verified": 1,
          },
        },
      ],
      "_type": "zoom_account",
      "active": true,
      "createdOn": undefined,
      "displayName": "JupiterOne Development",
      "email": "development@jupiterone.dev",
      "id": "H9Yx0-O8QUCXVIseYAsjaA",
      "name": "JupiterOne Development",
      "status": "active",
      "username": "development@jupiterone.dev",
    },
  ],
  "encounteredTypes": Array [
    "zoom_account",
  ],
  "numCollectedEntities": 1,
}
`;
